python/src/nanoarrow/meson.build

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+nanoarrow_proj = subproject(
+    'arrow-nanoarrow',
+    default_options: ['default_library=static']
+)
+nanoarrow_dep = nanoarrow_proj.get_variable('nanoarrow_dep')
+nanoarrow_ipc_dep = nanoarrow_proj.get_variable('nanoarrow_ipc_dep')
+nanoarrow_device_dep = nanoarrow_proj.get_variable('nanoarrow_device_dep')
+
+py = import('python').find_installation(pure: false)
+
+generated_pyx = custom_target(
+    'generate-pyx',
+    output: 'nanoarrow_c.pxd',
+    command: [py, meson.current_source_dir() + '/../../bootstrap.py'],
+)
+nanoarrow_c_dep = declare_dependency(sources: generated_pyx)
+
+cyfiles = [
+    '_array.pyx',
+    '_array_stream.pyx',
+    '_buffer.pyx',
+    '_device.pyx',
+    '_ipc_lib.pyx',
+    '_schema.pyx',
+    '_types.pyx',
+    '_utils.pyx',
+]
+
+cython_args = [
+    '--include-dir',
+    meson.current_build_dir(),
+    '--include-dir',
+    meson.current_source_dir(),
+    '--include-dir',
+    meson.source_root() / 'vendor',  # for generated nanoarrow_c file
+]
+if get_option('buildtype') == 'debug'
+    cython_args += ['--gdb']
+endif
+
+# Seems like some Cython files do #include "nanoarrow.h" instead of
+# #include "nanoarrow/nanoarrow.h"
+nanoarrow_src_dir = nanoarrow_proj.get_variable('_nanoarrow_src_dir')
+
+fs = import('fs')
+foreach cyf : cyfiles
+  py.extension_module(
+      fs.stem(cyf),
+      sources: [cyf],
+      cython_args: cython_args,
+      include_directories: nanoarrow_src_dir,
+      dependencies: [
```

Comment:
This is a little too broad to use every dependency for every Cython file; can refine so that only IPC files need the IPC dep, Device files need the device dep, etc...